pkg/webhook/statefulset/statefulset.go

```diff
@@ -68,22 +70,20 @@ func AdmitStatefulSets(ar v1beta1.AdmissionReview) *v1beta1.AdmissionResponse {
 		}
 	}
 
-	raw := ar.Request.OldObject.Raw
-	set := apps.StatefulSet{}
-	if _, _, err := deserializer.Decode(raw, nil, &set); err != nil {
-		err := fmt.Errorf("statefulset %s/%s, decode request failed, err: %v", namespace, name, err)
-		glog.Errorf(err.Error())
+	stsObjectMeta, stsPartition, err := getStsAttributes(ar.Request.OldObject.Raw, apiVersion)
+	if err != nil {
+		glog.Errorf("statefulset %s/%s, decode request failed, err: %v", namespace, name, err)
 		return util.ARFail(err)
```

Comment:
I think this return error is not meaningful, should return a more detailed error, just like the error output above.

Reply:
Done